Add field to rd_kafka_queue which denotes if it contains fetched msgs (#4256)
confluentinc/librdkafka: CHANGELOG.md

   calling `rd_kafka_message_leader_epoch()` on the polled `rkmessage` (#4245).
 * Fix a segmentation fault when fetching from follower and the partition lease
   expires while waiting for the result of a list offsets operation (#4254).

>  * Fixes certain cases where polling would not rejoin the consumer group

Reviewer: Not setting `polled` would also make it leave the consumer group I think.
```suggestion
 * Fixes certain cases where polling would not keep the consumer
   in the group or make it rejoin it (#4256).
```

>    when the cause of leaving the group was exceeding `max.poll.interval.ms`.

Reviewer: ```suggestion
```

   allowing threads different from the main one to call
   the `rd_kafka_toppar_set_fetch_state` function, given they hold
   the lock on the `rktp`.
 * In v2.1.0, a bug was fixed which caused polling any queue to reset the
   `max.poll.interval.ms`. Only certain functions were made to reset the timer,
   but it is possible for the user to obtain the queue with messages from
   the broker, skipping these functions. This was fixed by encoding information

>    in a queue itself, that whether polling it should reset the timer.

Reviewer: ```suggestion
   in a queue itself, that, whether polling, resets the timer.
```
